### PR TITLE
feat(stat-detectors): Add widget for surfacing geo location regressions

### PR DESCRIPTION
**Pull request by narsaynorath**
The widget makes a request to the advanced analysis endpoint and shows the top 5 results in list format. The right-most side calculates the percent change and links users to the transaction summary page with that country code prefiltered.

I chose to use the duration breakdown chart because it doesn't seem like the trends chart propagates the filter. I suspect this is because the trends endpoint is strictly metrics based so we eliminated the ability to filter in that chart.

Screenshot 2023-10-16 at 3 59 06 PM

Until we get flag icons, we'll just use the list format with the country code and country name for now.

### CODE REVIEW

**Review by malwilley — COMMENTED** [review]
Could you add some tests for the new component?

**Comment by narsaynorath** [comment]
> Could you add some tests for the new component?

@malwilley Added a number of tests. Specifically for rendering the data, the tooltip on hover, and the target URL state

File: static/app/components/events/eventStatisticalDetector/geoLocationDiff.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+  const PROJECT_ID = '1';
+
+  // Breakpoint is stored as seconds
+  const BREAKPOINT_TIMESTAMP = (Date.now() - 7 * DAY) / 1000;
+  let mockEvent: Event;
+
+  beforeEach(() => {
+    mockEvent = {
+      ...EventMock(),
+      occurrence: {
```
Comment (malwilley) [inline static/app/components/events/eventStatisticalDetector/geoLocationDiff.spec.tsx:21]: nit: can pass overrides into the fixture like this: `const mockEvent = EventMock({occurrence: {...}})`

**Review by malwilley — APPROVED** [review]
Nice tests! 💯

**Pull request closed without merging**